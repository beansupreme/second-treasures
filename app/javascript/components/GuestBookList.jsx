import React from "react";
import axios from "axios";
import GuestBookListItem from './GuestBookListItem'
class GuestBookList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      books: []
    }
  }

  componentDidMount() {
    axios.get('/api/v1/books').then(response => {
      this.setState({books: response.data})
    });
  }


  render () {
    const bookItems = this.state.books.map((book) => 
      <GuestBookListItem key={book.id} book={book}/>
    );
    return (
      <div id="recommended-book-list">
        {bookItems}
      </div>
    );
  }
}

export default GuestBookList
