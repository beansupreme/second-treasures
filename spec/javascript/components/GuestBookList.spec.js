import React from 'react';
import renderer from 'react-test-renderer';
import GuestBookList from 'components/GuestBookList';

describe('<GuestBookList />', () => {
  it('renders as expected', () => {
    const tree = renderer.create(<GuestBookList />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});