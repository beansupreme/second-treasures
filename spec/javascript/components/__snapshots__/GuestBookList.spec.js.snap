// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GuestBookList /> renders as expected 1`] = `
<div
  id="recommended-book-list"
/>
`;
